Rectangle/WindowCalculation/BottomHalfCalculation.swift

```diff
@@ -8,30 +8,37 @@
 
 import Foundation
 
+private enum SupportedPositions: Int {
+    case half = 0, twoThirds, oneThird
+}
+
 class BottomHalfCalculation: WindowCalculation {
     
-    func calculateRect(_ windowRect: CGRect, visibleFrameOfScreen: CGRect, action: WindowAction) -> CGRect? {
+    func calculateRect(_ windowRect: CGRect, lastAction: RectangleAction?, visibleFrameOfScreen: CGRect, action: WindowAction) -> CGRect? {
         
         var oneHalfRect = visibleFrameOfScreen
         oneHalfRect.size.height = floor(oneHalfRect.height / 2.0)
         
-        if Defaults.subsequentExecutionMode.value != .none {
-            if abs(windowRect.midX - oneHalfRect.midX) <= 1.0 {
+        if Defaults.subsequentExecutionMode.value == .none {
+            return oneHalfRect
+        }
+        
+        let count = (lastAction?.action == action) ? lastAction!.count : 0
```

Comment:
Perhaps I should write some coding guidelines for the project (you're the first one to encounter my nit picking). 

I never force unwrap. There is pretty much never a reason why I would want my app to crash if an optional is null. Even though that's not possible in this case, there may come a day when calculateRect gets refactored to take an optional WindowAction, and then we're wondering why it crashed somewhere down the line. As a user, I prefer having something just unexpectedly not happen in the app instead of having it crash seemingly randomly.

I think I would prefer this to look like this:
```
let count: Int? = lastAction?.action == action
    ? lastAction?.count
    : 0
```

Of course, then we can't make use of a switch statement below, but that's not a huge deal to me since we don't have a ton of cases.

Reply:
I originally had a a `if let` statement to unwrap etc., but then I rewrote it to one line as I didn't see how it was possible to throw an error from there as the isn't optional on the struct. However I see from reading online the swift compiler isn't that smart compared to others and if you use '!' the compiler won't double check and give you a warning.

Instead of using `var` what do you think of using the nil-coalescing operator to safely unwrap the optional?

`let count = (lastAction?.action == action) ? (lastAction?.count ?? 0) : 0`

Reply:
Sure, that works for me.